addons/pos_restaurant/static/src/app/bill_screen/bill_screen.js

```diff
@@ -1,29 +1,27 @@
 /** @odoo-module */
 
-import { ReceiptScreen } from "@point_of_sale/app/screens/receipt_screen/receipt_screen";
+import { Dialog } from "@web/core/dialog/dialog";
 import { OrderReceipt } from "@point_of_sale/app/screens/receipt_screen/receipt/order_receipt";
-import { registry } from "@web/core/registry";
+import { Component, useState } from "@odoo/owl";
+import { usePos } from "@point_of_sale/app/store/pos_hook";
+import { useService } from "@web/core/utils/hooks";
 
-export class BillScreen extends ReceiptScreen {
+export class BillScreen extends Component {
     static template = "pos_restaurant.BillScreen";
-    static components = { OrderReceipt };
-    confirm() {
-        if (!this.env.isMobile) {
-            this.props.resolve({ confirmed: true, payload: null });
-            this.pos.closeTempScreen();
-        }
+    static components = { OrderReceipt, Dialog };
+    setup() {
+        this.state = useState({ printing: false });
+        this.pos = usePos();
+        this.printer = useState(useService("printer"));
     }
-    /**
-     * @override
-     */
     async printReceipt() {
-        await super.printReceipt();
-        this.currentOrder._printed = false;
-    }
-
-    get isBill() {
-        return true;
+        this.pos.get_order()._printed = await this.printer.print(
+            OrderReceipt,
+            {
+                data: this.pos.get_order().export_for_printing(),
+                formatCurrency: this.env.utils.formatCurrency,
+            },
+            { webPrintFallback: true }
+        );
```

Comment:
why are you doing this? This is not part of the logic. `_printed` should only be true if the order is validated.

Reply:
indeed

Reply:
this code was the same in the bill screen as in the receipt screen so i lifted the function to the pos store

Reply:
that's true. Now it should have the same behavior as before